atom/common/promise_util.h

```diff
@@ -125,6 +167,47 @@ class CopyablePromise {
   CopyablePromise(const CopyablePromise&);
   ~CopyablePromise();
 
+  template <typename T>
+  static void ResolveCopyablePromise(const CopyablePromise& promise, T result) {
+    if (!content::BrowserThread::CurrentlyOn(content::BrowserThread::UI)) {
+      base::PostTaskWithTraits(
+          FROM_HERE, {content::BrowserThread::UI},
+          base::Bind([](const CopyablePromise& promise,
```

Comment:
This could bind to the ResolvePromise helpers declared above and use `std::move(promise.GetPromise())`. Can help eliminate this lamda.  Similarly for other binds below.

```
base::PostTaskWithTraits(
    FROM_HERE, {content::BrowserThread::UI},
    base::BindOnce(ResolvePromise, std::move(promise.GetPromise()), std::move(result)));
```